specs/SingleAppHost.md

```diff
@@ -0,0 +1,60 @@
+# Background
+While WebViews from different app processes normally don't share the same WebView browser process instance, app processes from the same product suite may share the same webview browser process
+instance by specifying the same user data folder when creating WebView2Environment object.
+The `IsForSingleAppHost` property is for the developer to express the sharing intent, so that we could provide optimized security and performance according expected usage.
+
+# Description
+The `IsForSingleAppHost` property indicates whether other processes can create WebView2 sharing the same WebView browser process instance by using WebView2Environment created with the same user data folder.
+Default is FALSE, meaning that other processes can create WebView objects sharing the same browser process instance.
+
+# Examples
+## Win32 C++
+```cpp
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_IsForSingleAppHost(TRUE);
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions(
+        nullptr, m_userDataFolder.c_str(), options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            this, &AppWindow::OnCreateEnvironmentCompleted).Get());
+```
+## WinRT and .NET
+```c#
+auto options = new CoreWebView2EnvironmentOptions();
+options.IsForSingleAppHost = true;
+auto environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+```
+
+# Remarks
+See API details.
+
+# API Details
+## Win32 C++
+```cpp
+interface ICoreWebView2EnvironmentOptions_2 : IUnknown {
+
+  /// Whether other processes can create WebView2 sharing the same WebView browser
+  /// process instance by using WebView2Environment created with the same user data folder.
+  /// Default is FALSE, meaning that other processes can create WebView objects
+  /// sharing the same browser process instance.
+  [propget] HRESULT IsForSingleAppHost([out, retval] BOOL* value);
```

Comment:
App is sort of less well defined notion but at least elsewhere in Windows can represent a collection of processes (like AUMID). In this case we're limiting to one process not app so we can say process here? 

Additionally instead of saying this is for a single app host we can make a stronger statement of whether a single host is required or more than one host is allowed.

Ideas:
* IsSingleHostProcessRequired - This seems weirdly worded to me. Not sure how to express this better describing it from the single host case.
* AreMultipleHostProcessesAllowed - This seems to more accurately describe the property.

Reply:
Changed to AreMultipleHostProcessesAllowed. That is a much better name.